packages/mobile/src/components/expandable/Expandable.tsx

```diff
@@ -0,0 +1,70 @@
+import type { ReactNode } from 'react'
+import { useState, useCallback } from 'react'
+
+import type { StyleProp, ViewStyle } from 'react-native'
+import { Animated, LayoutAnimation, TouchableOpacity, View } from 'react-native'
+
+import { makeStyles } from 'app/styles'
+
+const useStyles = makeStyles(({ palette, spacing, typography }) => ({
+  root: {
+    display: 'flex'
+  }
+}))
+
+const springToValue = ({
+  animation,
+  value
+}: {
+  animation: Animated.Value
+  value: number
+}) => {
+  Animated.spring(animation, {
+    toValue: value,
+    tension: 160,
+    friction: 15,
+    useNativeDriver: true
+  }).start()
+}
+
+export const useExpandable = () => {
+  const [isExpanded, setIsExpanded] = useState(false)
+  return { isExpanded, setIsExpanded, springToValue }
+}
+
+type ExpandableProps = {
+  isExpanded: boolean
+  setIsExpanded: (param: (expanded: boolean) => boolean) => void
+  children?: ReactNode
+  renderHeader?: () => ReactNode
+  onExpand?: () => void
+  style?: StyleProp<ViewStyle>
+}
+
+export const Expandable = ({
+  isExpanded,
+  setIsExpanded,
+  children,
+  onExpand,
+  renderHeader,
+  style: styleProp
+}: ExpandableProps) => {
```

Comment:
I think this would be simpler as a standard controlled component. As it is, I'm not sure there's a need to export the hook and spring behavior.
I think you could just have this component accept `isExpanded` (or even better just `expanded`) and also a callback for when the user clicks the expander (`onToggleExpansion`). Then have the parent component implement a simple setState toggle pattern (`handleToggleExpansion = setExpanded(expanded => !expanded)`).

And this component would use an effect watching the expansion state and trigger the spring/layoutanimation stuff whenever we change states.